test/integration/Morpho.t.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+import {Test} from 'forge-std/Test.sol';
+import {IERC20} from 'lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol';
+import {Router, IRouter} from 'src/Router.sol';
+import {IAgent} from 'src/interfaces/IAgent.sol';
+import {DataType} from 'src/libraries/DataType.sol';
+import {Test} from 'forge-std/Test.sol';
+import {MorphoFlashLoanCallback} from 'src/callbacks/MorphoFlashLoanCallback.sol';
+
+interface IMorphoFlashLoanCallback {
+    function flashLoan(address token, uint256 assets, bytes calldata data) external;
+}
+
+contract MorphoIntegrationTest is Test {
+    // using SafeERC20 for IERC20;
+
+    address public constant MORPHO = 0x64c7044050Ba0431252df24fEd4d9635a275CB41;
+    IERC20 public constant USDC = IERC20(0x62bD2A599664D421132d7C54AB4DbE3233f4f0Ae);
+
+    address public user;
+    IRouter public router;
+    IAgent public agent;
+    MorphoFlashLoanCallback public flashLoanCallback;
+
+    // Empty arrays
+    address[] public tokensReturnEmpty;
+    DataType.Input[] public inputsEmpty;
+    bytes[] public permit2DatasEmpty;
+
+    function setUp() external {
+        vm.createSelectFork(vm.rpcUrl('goerli'), 10310460);
+
+        user = makeAddr('User');
+        router = new Router(makeAddr('WrappedNative'), makeAddr('Permit2'), address(this));
+        vm.prank(user);
+        agent = IAgent(router.newAgent());
+        flashLoanCallback = new MorphoFlashLoanCallback(address(router), MORPHO, 0);
+
+        vm.label(address(router), 'Router');
+        vm.label(address(agent), 'Agent');
+        vm.label(MORPHO, 'Morpho');
+        vm.label(address(flashLoanCallback), 'MorphoFlashLoanCallback');
+        vm.label(address(USDC), 'USDC');
+    }
+
+    function testExecuteMorphoFlashLoan(uint256 amount) external {
+        IERC20 borrowedToken = USDC;
+        address flashloanPool = MORPHO;
```

Comment:
```suggestion
        address flashLoanPool = MORPHO;
```
However, I suggest to use `MORPHO` directly as all tokens are in this singleton contract even we change `USDC` to other token afterward.